src/components/Table/EditColumnsButton.tsx

```diff
@@ -0,0 +1,78 @@
+import { Dispatch, SetStateAction, useRef } from "react";
+import { useMenuTrigger } from "react-aria";
+import { useMenuTriggerState } from "react-stately";
+import { Checkbox } from "src/inputs";
+import { Css, px } from "../../Css";
+import { useTestIds } from "../../utils";
+import { Button } from "../Button";
+import { isIconButton, isTextButton, OverlayTrigger, OverlayTriggerProps } from "../internal/OverlayTrigger";
+import { GridColumn, Kinded } from "./GridTable";
+import { Columns } from "./useColumns";
+
+interface EditColumnsButtonProps<R extends Kinded, S>
+  extends Pick<OverlayTriggerProps, "trigger" | "placement" | "disabled" | "tooltip"> {
+  columns: GridColumn<R, S>[];
+  setColumns: Dispatch<SetStateAction<Columns<R, S>>>;
+  title?: string;
+  buttonText?: string;
+  // for storybook purposes
+  defaultOpen?: boolean;
+}
+
+export function EditColumnsButton<R extends Kinded, S = {}>(props: EditColumnsButtonProps<R, S>) {
+  const { defaultOpen, disabled, columns, setColumns, trigger, title, buttonText } = props;
+  const state = useMenuTriggerState({ isOpen: defaultOpen });
+  const buttonRef = useRef<HTMLButtonElement>(null);
+  const { menuTriggerProps } = useMenuTrigger({ isDisabled: !!disabled }, state, buttonRef);
+  const tid = useTestIds(
+    props,
+    isTextButton(trigger) ? trigger.label : isIconButton(trigger) ? trigger.icon : trigger.name,
+  );
+
+  function clearSelections() {
+    columns.map((column) => (column.visible = false));
+    setColumns((prevState) => ({
+      ...prevState,
+      visibleColumns: columns.filter((column) => (column.canHide && column.visible) || !column.canHide),
+    }));
+  }
+
+  return (
+    <OverlayTrigger {...props} menuTriggerProps={menuTriggerProps} state={state} buttonRef={buttonRef} {...tid}>
+      <div
+        css={{
+          ...Css.df.fdc.bgWhite.py3.px2.maxw(px(380)).$,
+          "&:hover": Css.bshHover.$,
+        }}
+      >
+        <div css={Css.gray500.sm.ttu.$}>{title || "Select columns to show"}</div>
+        <div css={Css.df.add({ flexWrap: "wrap" }).gap2.py2.$}>
+          {columns.map((column, i) => {
+            return (
+              column.canHide && (
+                <div css={Css.add({ flex: "40%" }).ttc.$} key={i}>
+                  <Checkbox
+                    label={column.name || ""}
```

Comment:
**thought:** I wonder if we should do a `console.warn` if the column does not have a `name` property. Also, rather than showing a blank `label`, I think we should filter out columns without a name.
So maybe ahead of this loop we do something along the lines of:
```
const editableColumns = useMemo(() => columns.filter((c) => {
  if (!c.canHide) return false;
  if (!c.name || c.name.length === 0) { 
    console.warn("Column is missing 'name' property required by the Edit Columns button", column);
    return false;
  }
  return true;
}), [columns])
```
(pseudo code, so please don't expect it to immediately work 😄)